internal/services/groups/groups_data_source_test.go

```diff
@@ -121,3 +135,12 @@ data "azuread_groups" "test" {
 }
 `
 }
+
+func (GroupsDataSource) securityEnabled() string {
+	return `
+data "azuread_groups" "test" {
+  return_all       = true
+  security_enabled = true
```

Comment:
These should be mutually exclusive since `return_all` implies all groups? Alternatively we could update the docs to reflect that `return_all` excludes `display_names` and `object_ids` but can be used with `security_enabled` to return all security-enabled groups.

Reply:
I think I agree with the alternative. Otherwise we can look to apply a number of filters for other fields at the same time.

Reply:
Sounds good to me!